src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp

```diff
@@ -114,91 +114,136 @@ struct LinearSolverParameters
-      jacobi,    ///< Jacobi
-      l1jacobi,  ///< l1-Jacobi
-      fgs,       ///< Gauss-Seidel (forward sweep)
-      sgs,       ///< Symmetric Gauss-Seidel
-      l1sgs,     ///< l1-Symmetric Gauss-Seidel
-      chebyshev, ///< Chebyshev polynomial
-      direct,    ///< Direct solver as preconditioner
-      bgs,       ///< Gauss-Seidel smoothing (backward sweep)
+      default_,           ///< Use LAI's default option
+      jacobi,             ///< Jacobi (GPU support in hypre)
+      l1jacobi,           ///< l1-Jacobi (GPU support in hypre)
+      fgs,                ///< Gauss-Seidel (forward sweep)
+      sgs,                ///< Symmetric Gauss-Seidel
+      l1sgs,              ///< l1-Symmetric Gauss-Seidel
+      chebyshev,          ///< Chebyshev polynomial (GPU support in hypre)
+      direct,             ///< Direct solver as preconditioner
+      bgs                 ///< Gauss-Seidel smoothing (backward sweep)
+    };
+
+    /// AMG coarsening types (HYPRE only)
+    enum class CoarseningType : integer
+    {
+      default_,           ///< Use LAI's default option
+      CLJP,               ///< A parallel coarsening algorithm using independent sets
+      RugeStueben,        ///< Classical Ruge-Stueben on each processor, followed by a third pass
+      Falgout,            ///< Ruge-Stueben followed by CLJP
+      PMIS,               ///< Parallel coarsening as CLJP but with lower complexities (GPU support)
+      HMIS                ///< Hybrid PMIS coarsening
+    };
+
+    /// AMG interpolation type (HYPRE only)
+    enum class InterpType : integer
+    {
+      default_,           ///< Use LAI's default option
+      modifiedClassical,  ///< Modified classical
+      direct,             ///< Direct (GPU support)
+      multipass,          ///< Multipass (GPU support)
+      extendedI,          ///< Extended+i (GPU support)
+      standard,           ///< Standard
+      extended,           ///< Extended classical (GPU support)
+      directBAMG,         ///< Direct with separation of weights (GPU support)
+      modifiedExtended,   ///< Modularized extended classical (GPU support)
+      modifiedExtendedI,  ///< Modularized extended+i (GPU support)
+      modifiedExtendedE   ///< Modularized extended+e (GPU support)
+    };
+
+    /// AMG interpolation type for aggressive coarsening levels (HYPRE only)
+    enum class AggInterpType : integer
+    {
+      default_,           ///< Use LAI's default option
+      extendedIStage2,    ///< Extended+i 2-stage (GPU support)
+      standardStage2,     ///< Standard 2-stage
+      extendedStage2,     ///< Extended 2-stage (GPU support)
+      multipass,          ///< Multipass (GPU support)
+      modifiedExtended,   ///< Modularized Extended (GPU support)
+      modifiedExtendedI,  ///< Modularized Extended+i (GPU support)
+      modifiedExtendedE,  ///< Modularized Extended+e (GPU support)
+      modifiedMultipass   ///< Modularized Multipass (GPU support)
```

Comment:
I personally like spelled-out names. The code becomes very readable.

@rrsettgast @joshua-white what do you think?

Reply:
I agree